### PR TITLE
RSDK-6465 [part 2]: use helper functions in sensor controlled base

### PR DESCRIPTION
**Pull request by martha-johnston**

### CODE REVIEW

File: components/base/sensorcontrolled/velocities.go [hunk 1]
```diff
@@ -36,85 +35,68 @@ func (sb *sensorBase) setupControlLoops() error {
...
+func (sb *sensorBase) setupControlLoop(linear, angular control.PIDConfig) error {
+	// set the necessary options for a sensorcontrolled base
+	options := control.Options{
+		SensorFeedback2DVelocityControl: true,
+		LoopFrequency:                   20,
+		ControllableType:                "base_name",
+	}
+
+	// check if either linear or angular need to be tuned
+	if (linear.P == 0.0 && linear.I == 0.0 && linear.D == 0.0) ||
```
Comment (JohnN193) [inline components/base/sensorcontrolled/velocities.go:47]: [opt] we could make this check a method on PIDConfig. Would let us set the autoconfig flag by using 

`options.NeedsAutoTuning = linear.NeedsTuning || angular.NeedsTuning`

though theres definitely an argument to leave these checks as is so the conditions are explicit.

File: components/base/sensorcontrolled/velocities.go [hunk 2]
```diff
@@ -36,85 +35,68 @@ func (sb *sensorBase) setupControlLoops() error {
...
 	// set linear setpoint config
 	linConf := control.BlockConfig{
-		Name: "linear_setpoint",
+		Name: sb.blockNames["constant"][0],
 		Type: "constant",
 		Attribute: rdkutils.AttributeMap{
 			"constant_val": linearValue,
 		},
 		DependsOn: []string{},
 	}
```
Comment (JohnN193) [inline components/base/sensorcontrolled/velocities.go:79]: [opt] make a function to construct a new BlockConfig

**Review by JohnN193 — APPROVED** [review]
LGTM, just some opts for more helper functions

File: components/base/sensorcontrolled/velocities.go [hunk 3]
```diff
@@ -17,12 +16,12 @@ import (
 // is limited to 8 bits, or the range 0-255.
 const rPiGain = 0.00392157
```
Comment (susmitaSanyal) [inline components/base/sensorcontrolled/velocities.go:17]: Seems like you are not using rPiGain anymore, might wanna remove this.

**Review by susmitaSanyal — APPROVED** [review]
LGTM! just one nit.

File: components/base/sensorcontrolled/velocities.go (hunk 1 above)
Comment (martha-johnston) [reply components/base/sensorcontrolled/velocities.go:47]: hmm I'm not opposed, I'll try it out see if it makes sense

Comment (martha-johnston) [reply components/base/sensorcontrolled/velocities.go]: need to test but I think what I added should work

**Comment by martha-johnston** [comment]
just tested and auto-tuning works with these new changes

**Pull request merged**